src/Coverage.jl

```diff
@@ -232,27 +246,80 @@ module Coverage
     child folders. Default folder is "src", which is useful for the primary case
     where Coverage is called from the root directory of a package.
     """
-    function process_folder(folder="src")
+    function process_folder(folder="src", exclusions_file=".coverage.yml")
         @info "Coverage.process_folder: Searching $folder for .jl files..."
         source_files = FileCoverage[]
         files = readdir(folder)
+        excl_file = joinpath(folder, exclusions_file)
+        excluded_files, excluded_line_patterns = get_exclusions(excl_file)
+
         for file in files
+            if file in excluded_files
+                @info "Coverage.process_folder: Skipping $file, in exclusion list"
+                continue
+            end
+
             fullfile = joinpath(folder, file)
             if isfile(fullfile)
                 # Is it a Julia file?
                 if splitext(fullfile)[2] == ".jl"
-                    push!(source_files, process_file(fullfile, folder))
+                    push!(source_files, process_file(fullfile, folder, excluded_line_patterns))
                 else
                     @debug "Coverage.process_folder: Skipping $file, not a .jl file"
                 end
             elseif isdir(fullfile)
                 # If it is a folder, recursively traverse
+                # note: each folder should have it's own .coverage.yml
                 append!(source_files, process_folder(fullfile))
             end
         end
         return source_files
     end
 
+    """
+        get_exclusions(file::AbstractString) -> (excluded_files, excluded_line_patterns)
+
+    Parse the provided file as a coverage exclusions file (usually named
+    ``.coverage.yml`). This file should have a format like the following:
+    ```
+    # exclude files in the directory with these names
+    exclude_files:
+      - foo.jl
+      - bar.jl
+
+    # exclude any lines that match any of the following regexes
+    exclude_patterns:
+      - "[NO COVERAGE]"
```

Comment:
Hmm, so if I want to use this, I have to keep remember to put such a file in every single directory containing source code? If the exclusion file applied recursively, wouldn't that be more convenient (and close to how e.g. `.gitignore` works) ?

Or, put another way: Personally, I'd be content with a single master config file in the repository root; I can see imagine that some people might want to be able to override it for some subdirs (though I can't imagine *why*, but that just shows my lack of imagination :-). But having to put this file in every subdir? I am sure I'd forget.

Mind you, this is certainly not very critical, nor a deal breaker. I just wonder about the design choice.

Reply:
That is for me also the biggest design question (see my top comment), and making it recursive is probably the best. I'll work on changing this.

Reply:
Done.